### PR TITLE
docs: onboarding docs

### PR DESCRIPTION
**Pull request by Birkbjo**
This is meant to provide some background and introduction to the architecture of the App Hub. It's based on my notes for the App Hub-onboarding meeting, so I don't know if the format is ideal but I would like for it to be a bridge/mix between "high-overview" and technical as to provide insight while being practical for a hands-on onboarding.

### CODE REVIEW

File: onboarding.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+`parseDatabaseJson()` is meant to be a function that can be overriden for more specific transformations, and will be called with the result of a database-query and is meant to return the data for internal application use. `formatDatabaseJson` is the opposite - it takes data from the application and transforms it to database-friendly objects.
+
+
+Many would think that services and the models should be classes, and I've been a bit back and forth. I'm deliberately not using classes unless it's needed, and most of the time that means when you need state. Up to now these do not need state, but it could make sense to have a base-class that lays out the structure of these modules. But classes needs to instantiated, and that would mean bootstrapping the services on startup which is annoying in itself, and the methods would also not be statically available. [Schmervice](https://github.com/hapipal/schmervice) is also something we could consider.
```
Comment (Birkbjo) [inline onboarding.md:58]: Should this paragraph be left out? I've multiple reflections like this that I want to get out and would like feedback on, but I'm unsure if they are just bloat here.

Comment (amcgee) [reply onboarding.md:58]: I think it's fine, we can always clean it up and remove info later!

**Review by amcgee — APPROVED** [review]
Looks good to me - @Birkbjo maybe we should move this to the more conventional `CONTRIBUTING.md` instead of `onboarding.md`?

**Comment by Birkbjo** [comment]
> Looks good to me - @Birkbjo maybe we should move this to the more conventional `CONTRIBUTING.md` instead of `onboarding.md`?

Hmm, I would expect contributing to be more like some general guidelines and methodology, than technical documentation like this, no?

**Comment by amcgee** [comment]
@Birkbjo went ahead and merged this, open to improvements (since it did say WIP)

**Pull request merged**